sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -216,7 +216,7 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
       val finalOptions = sessionOptions ++ extraOptions.toMap ++ pathsOption
       val dsOptions = new CaseInsensitiveStringMap(finalOptions.asJava)
       val table = userSpecifiedSchema match {
-        case Some(schema) => provider.getTable(dsOptions, schema)
+        case Some(schema) => provider.getTable(dsOptions, schema, Array.empty)
```

Comment:
This isn't correct. The `DataFrameReader` does not know that the table is unpartitioned.

Reply:
yeah, this is one of the most annoying and confusing behaviors of DataSourceV1. Being able to provide a schema but not partitioning information, which leads to minutes of partition schema inference.